3_bidirectional_cxx_interop/cmake/modules/AddSwift.cmake

```
@@ -23,27 +37,33 @@ function(_swift_generate_cxx_header_target target module header)
     set(SDK_FLAGS "-sdk" "${CMAKE_OSX_SYSROOT}")
   elseif(WIN32)
     set(SDK_FLAGS "-sdk" "$ENV{SDKROOT}")
+  elseif(DEFINED ${CMAKE_SYSROOT})
+    set(SDK_FLAGS "-sdk" "${CMAKE_SYSROOT}")
   endif()
 
-  add_custom_command(
-    OUTPUT
-      "${header}"
+  cmake_path(APPEND CMAKE_CURRENT_BINARY_DIR include
+    OUTPUT_VARIABLE base_path)
+
+  cmake_path(APPEND base_path ${header}
+    OUTPUT_VARIABLE header_path)
+
+  set(_AllSources $<TARGET_PROPERTY:${target},SOURCES>)
+  set(_SwiftSources $<FILTER:${_AllSources},INCLUDE,\\.swift$> )
+  add_custom_command(OUTPUT ${header_path}
+    DEPENDS ${_SwiftSources}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
     COMMAND
       ${CMAKE_Swift_COMPILER} -frontend -typecheck
       ${ARG_SEARCH_PATHS}
-      ${ARG_SOURCES}
+      ${_SwiftSources}
       ${SDK_FLAGS}
-      -module-name "${module}"
+      -module-name "${ARG_MODULE_NAME}"
       -cxx-interoperability-mode=default
-      -emit-clang-header-path "${header}"
-    DEPENDS
-      ${ARG_DEPENDS}
-    COMMENT
-      "Generating '${header}'"
-  )
-
-  add_custom_target("${target}"
-    DEPENDS
-      "${header}"
-  )
+      -emit-clang-header-path ${header_path})
```

If `_SwiftSources` contains multiple files, adding `COMMAND_EXPAND_LISTS` fixes #12 

```suggestion
      -emit-clang-header-path ${header_path}
      COMMAND_EXPAND_LISTS)
```

---

Ah, yes, good catch.